lib/spack/spack/test/conftest.py

```diff
@@ -1027,10 +1034,14 @@ def mock_archive(request, tmpdir_factory):
                                      ['url', 'path', 'archive_file',
                                       'expanded_archive_basedir'])
     archive_file = str(tmpdir.join(archive_name))
+    if sys.platform == 'win32':
+        url = ('file:///' + archive_file)
```

Comment:
(question) what fails without this distinction?

Reply:
This is redundant with to https://github.com/spack/spack/pull/27021#discussion_r796180739 so I think only one of those needs an answer (I'll leave both up though in case one provides more context).

Reply:
This actually answers a lot of questions I was having doing the path handling... I was wondering where that spare `/` was coming from. A number of fetching related tests fail when this is removed, seemingly due to how urllib is parsing the file urls. 
```
Archive(url='file://C:\\Users\\john.parent\\AppData\\Local\\Temp\\pytest-of-john.parent\\pytest-18\\mock-archive-dir0\...al\\Temp\\pytest-of-john.parent\\pytest-18\\mock-archive-dir0\\spack-src.tar.gz', expanded_archive_basedir='spack-src')
```
This filepath results in a `spack.fetch_strategy.FetchError: All fetchers failed for spack-stage-thpago2o` whereas the additional forward slash causes the test to pass.